extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -14,6 +25,70 @@ export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private @const {?../../amp-story/1.0/amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+
+    /** @private {?../../../src/service/localization.LocalizationService} */
+    this.localizationService_ = null;
+  }
+
+  /**
+   * @private
+   * @return {?Element} the parent amp-story-page
+   */
+  getPage_() {
+    return closest(
+      dev().assertElement(this.element),
+      (el) => el.tagName.toLowerCase() === 'amp-story-page'
+    );
+  }
+
+  /**
+   * Updates the CTA based on the shopping data.
+   * @private
+   */
+  updateCtaText_() {
+    const pageIdElem = this.getPage_();
+
+    const shoppingTags = Array.from(
+      pageIdElem.getElementsByTagName('amp-story-shopping-tag')
+    ).filter(
+      (tag) =>
+        this.storeService_.get(StateProperty.SHOPPING_DATA)[
+          tag.getAttribute('data-tag-id')
+        ]
+    );
+    const numShoppingTags = shoppingTags.length;
+
+    const ctaButton = pageIdElem.querySelector(
+      '.i-amphtml-story-page-open-attachment-host'
+    )?.shadowRoot.children[1];
+
+    if (numShoppingTags === 0 || !ctaButton) {
+      // Failsafe in case the CTA button fails to load, will call the function again until CTA button loads.
+      setTimeout(() => this.updateCtaText_(), 100);
+      return;
+    }
+
+    const getI18nString = (stringKey) =>
+      this.localizationService_.getLocalizedString(stringKey);
+
+    const i18nString =
+      numShoppingTags === 1
+        ? getI18nString(LocalizedStringId_Enum.AMP_STORY_SHOPPING_SHOP_LABEL) +
```

Comment:
We have a `localize` helper we can use. 
Please use this, we can then remove the localizationservice code from this PR.
https://github.com/ampproject/amphtml/blob/556e94a3057960a613a9349926a088a31f2874a5/extensions/amp-story/1.0/amp-story-unsupported-browser-layer.js#L22-L24

My bad for not suggesting this at first. It's brand new. I'm updating the codelab now.